src/loadScript.test.ts

```diff
@@ -0,0 +1,66 @@
+import { loadScript } from './loadScript';
+
+const goodURL = 'good-url';
+const badURL = 'bad-url';
+
+const getTestScript = (url: string) =>
+	Array.from(document.scripts).find(({ src }) => src.includes(url));
+
+const fakeLoadEvent = new MutationObserver((mutations) => {
+	if (
+		mutations.some((mutation) =>
+			Array.from(mutation.addedNodes).some(
+				({ nodeName }) => nodeName === 'SCRIPT',
+			),
+		)
+	) {
+		getTestScript(goodURL)?.onload?.(new Event('fake-onload-event'));
+		getTestScript(badURL)?.onerror?.(new Event('fake-onload-event'));
+	}
+});
+
+beforeEach(() => {
+	document.body.innerHTML = '';
+	document.body.appendChild(document.createElement('script'));
+	fakeLoadEvent.observe(document.body, {
+		childList: true,
+	});
+});
+
+afterEach(() => {
+	fakeLoadEvent.disconnect();
+});
+
+describe('loadScript', () => {
+	it('adds a script to the page and resolves the promise it returns when the script loads', async () => {
+		expect(document.scripts).toHaveLength(1);
+		await expect(loadScript(goodURL)).resolves.toMatchObject({
+			type: 'fake-onload-event',
+		});
+		expect(document.scripts).toHaveLength(2);
+	});
+
+	it('resolves immediately if a script with matching src is already on page and stops there', async () => {
+		expect(document.scripts).toHaveLength(1);
+		await expect(loadScript(goodURL)).resolves.toMatchObject({
+			type: 'fake-onload-event',
+		});
+		await expect(loadScript(goodURL)).resolves.toBeUndefined();
+		await expect(loadScript(goodURL)).resolves.toBeUndefined();
+		await expect(loadScript(goodURL)).resolves.toBeUndefined();
```

Comment:
Why 3x?

Reply:
just a random amount of repetition to make sure it doesn't get added. i'll add a comment